### PR TITLE
Dspec inline definition - fixes and improvements

### PR DESCRIPTION
**Pull request by lanarimarco**
## Description
Fixed issue where in the ELSE and ELSEIF branches the reference defined inline is ignored.

Added support of dspec inline definition for:
- CAT
- MOVE
- MOVEA
- SCAN
- MULT
- DIV
- DO
- SELECT

Fixed parsing error in case of level indicator in statements containing opcode PARM

## Checklist:
- [X] There are tests regarding this feature
- [X] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [X] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt [hunk 1]
```diff
@@ -84,11 +85,12 @@ interface CompositeStatement {
     val body: List<Statement>
 }
 
-fun List<Statement>.explode(): List<Statement> {
+fun List<Statement>.explode(preserveCompositeStatement: Boolean = false): List<Statement> {
```
Comment (FiorenzaBusi) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt:88]: what is the scope of `explode`?
Which type of statement is `preserveCompositeStatement`?

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/facade/RpgParserFacade.kt [hunk 2]
```diff
@@ -302,6 +302,7 @@ class RpgParserFacade {
         val parserResult: RpgParserResult
         val errors = LinkedList<Error>()
         val copyBlocks: CopyBlocks? = if (MainExecutionContext.getConfiguration().options.mustCreateCopyBlocks()) CopyBlocks() else null
+        // val includedCopySet = mutableSetOf<CopyId>()
```
Comment (FiorenzaBusi) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/facade/RpgParserFacade.kt:305]: When can comments be left and when not?

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt [hunk 3]
```diff
@@ -820,7 +820,7 @@ internal fun Cspec_fixed_standardContext.toAst(conf: ToAstConfiguration = ToAstC
     }
 }
 
-private fun Cspec_fixed_standard_partsContext.validate(stmt: Statement, conf: ToAstConfiguration): Statement {
+internal fun Cspec_fixed_standard_partsContext.validate(stmt: Statement, conf: ToAstConfiguration): Statement {
```
Comment (FiorenzaBusi) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt:823]: Why did you change private to internal?

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/statements.kt [hunk 4]
```diff
@@ -106,8 +92,63 @@ internal fun RpgParser.SelectstatementContext.toAst(conf: ToAstConfiguration = T
...
+    val dataDefinition = beginselect().csSELECT().cspec_fixed_standard_parts().toDataDefinition(result, position, conf)
+    return SelectStmt(
+        cases = whenClauses,
+        other = other,
+        dataDefinition = dataDefinition,
+        position = toPosition(conf.considerPosition)
+    )
+}
+
+internal fun RpgParser.BegindoContext.toAst(
```
Comment (FiorenzaBusi) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/statements.kt:106]: Why did you import `com.smeup.rpgparser.RpgParser.BlockContext` and in this case you didn't import `RpgParser.BegindoContext` ?

**Review by FiorenzaBusi — APPROVED** [review]
Ok

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt (hunk 1 above)
Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt:88]: There are two types of statements: Statement and CompositeStatement. 
CompositeStatement is a statement (for example IF) that contains other statements  (children statements contained in the  ELSE, ELSEIF, etc).
The function explode is used to extract the children statements but does not preserve CompositeStatement, for this reason I had to add param preserveCompositeStatement in order to preserve it. This thing has been helpful in the context of dspec inline definition for DO codop.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/facade/RpgParserFacade.kt (hunk 2 above)
Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/facade/RpgParserFacade.kt:305]: There are no rules, in that case I was not too sure about the feature I was implementing

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/statements.kt (hunk 4 above)
Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/statements.kt:106]: Should you ask it to intellij... In general I don't care
